Shopify/cli
# Enforce that all commands load an app

### WHY are these changes introduced?
All commands should track all app-related details.
To enforce that, we make every app command return an `AppInterface`, which means that an app was loaded at some point.



### WHAT is this pull request doing?
- Add a new ESLint rule to make all commands in `app/commands` extend `AppCommand`
- In `AppCommand` define the `run` function to return an object with an `AppInterface`, TS will force all child classes to use the same return type.
- Solve all the issues created by the two rules above: all commands extend `AppCommand`, all return an `AppInterface` now.

IN A FUTURE PR:
- All app commands should return a Session, to force authentication in all of them.
- Since all commands have auth + app loading -> extract that logic to a single place: `loadAppContext`


### How to test your changes?
Nothing to test in the CLI itself, you can try to modify the code of a command and see that the rules are being applied.


### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
